### DISCUSSION TITLE
Problemas com componente Visibility()

### DISCUSSION THREAD

**Discussion by lcsteil** [Flutterando/forum#105] [Perguntas e Respostas]
Boa noite,

Sou novo com flutter e estou com problemas para trabalhar com wgt Visibility().

Estou utilizando esse wgt para exibir o floatingActionButton

`floatingActionButton: Visibility(
          visible: _showFilter,
          child: FloatingActionButton(
            onPressed: _filter,
            child: Icon(
              Icons.filter_list,
              color: Colors.white,
              size: 24.0,
            ),
          ),`

Estou utilizando o TRIPLE, e preciso exibir o botão de acordo com o resultado da store.

Há tentei setar _showFilter para TRUE dentro do ScopedBuilder, mas não exibe o botão. Exibe somente quando eu salvo o arquivo.

Quando tento usar o setState dentro do ScopedBuilder dá erro.

Agradeço desde já!

**Comment by 30103010** [comment]
como esta a funcao que seta o seu _sowFilter pra True ?
o certo é usar um setstate dentro dessa funcao. exemplo:

void alteraExibicao() {
    setState(() {
      _sowFilter = !_sowFilter;
    });
  }

essa funcao tem que estar dentro da classe que voce quer atualizar , nao precisa estar dentro do  builder

**Reply by lcsteil** [reply]
Bom dia!

Eu já cheguei a fazer dessa forma.

O problema é que na store eu consumo uma API e no retorno, dependendo do que retornar eu quero exibir o botão....

A minha dúvida é onde chamar essa função.....

Obrigado pelo retorno!

**Reply by 30103010** [reply]
faz assim , na funcao que voce consome a API , passa o setstate da sua tela ... exemplo:

digamos que vc tem sua tela que deve ser atualizada.
cria uma funcao nela assim: 

atualizar() {
    setState(() {});
}

ai na funcao que vc chama a API vc cria um parametro do tipo function

Future<void> executaApi(bool executaAPI, Function suaFuncao) async {
  //api que ser executada qui
  if (executaAPI) {
    //api sera executada
    //se o retorno for o que vc esperava
    suaFuncao();
  }
}

vc chama essa funcao executaAPI na tela principal , mesmo ela rodando em outra classe , ela vai executar essa "suaFuncao" la na tela que vc chamou

**Reply by 30103010** [reply]
ai vc chama a funcao externa 

executaAPI(true, atualizar);

testa ai .. ve se da certo

**Reply by 30103010** [reply]
adiciona meu whatsapp qualquer coisa (17)98145-1701

**Reply by lcsteil** [reply]
O problema é que estou usando o triple....

`class detailsStore
    extends NotifierStore<Exception, detailModel>`

Ele não deixa eu setar um state na store......

**Comment by eduardoflorence** [comment]
Olá @lcsteil,
1. Se você está utilizando `ScopeBuilder` não é para utilizar `setState`, pois o próprio `ScopeBuilder` vai acionar a reatividade
2. Se você está utilizando `ScopeBuilder`, sua classe `DetailStore` não deve estender `NotifierStore` e sim `StreamStore`

Dá uma lida na documentação aqui: https://triple.flutterando.com.br/docs/getting-started/using-flutter-triple/
